ouroboros-consensus-protocol/src/ouroboros-consensus-protocol/Ouroboros/Consensus/Protocol/Ledger/HotKey.hs

```diff
@@ -148,60 +183,126 @@ sign = sign_
 -- | The actual KES key, unless it expired, in which case it is replaced by
 -- \"poison\".
 data KESKey c =
-    KESKey !(SL.SignKeyKES c)
+    KESKey !(OCert.OCert c) !(SL.SignKeyKES c)
   | KESKeyPoisoned
   deriving (Generic)
 
 instance Crypto c => NoThunks (KESKey c)
 
 kesKeyIsPoisoned :: KESKey c -> Bool
 kesKeyIsPoisoned KESKeyPoisoned = True
-kesKeyIsPoisoned (KESKey _)     = False
+kesKeyIsPoisoned (KESKey _ _)     = False
 
 data KESState c = KESState {
-      kesStateInfo :: !KESInfo
-    , kesStateKey  :: !(KESKey c)
+      kesStateInfo  :: !KESInfo
+    , kesStateKey   :: !(KESKey c)
     }
   deriving (Generic)
 
 instance Crypto c => NoThunks (KESState c)
 
+-- Create a new 'HotKey' and initialize it to the given initial KES key. The
+-- initial key must be at evolution 0 (i.e., freshly generated and never
+-- evolved).
 mkHotKey ::
      forall m c. (Crypto c, IOLike m)
-  => SL.SignKeyKES c
+  => OCert.OCert c
+  -> SL.SignKeyKES c
+  -> Absolute.KESPeriod  -- ^ Start period
+  -> Word64              -- ^ Max KES evolutions
+  -> m (HotKey c m)
+mkHotKey ocert initKey startPeriod maxKESEvolutions = do
+  hotKey <- mkEmptyHotKey maxKESEvolutions (pure ())
+  set hotKey ocert initKey 0 startPeriod
+  return hotKey
+
+-- Create a new 'HotKey' and initialize it to the given initial KES key. The
+-- initial key should be at the given evolution.
+mkHotKeyEv ::
+     forall m c. (Crypto c, IOLike m)
+  => Word
```

Comment:
```suggestion
mkHotKeyAtEvolution ::
     forall m c. (Crypto c, IOLike m)
  => Evolution
```

Could be worth creating a `newtype Evolution = Evolution Word`

Reply:
Yes, but that would have to live in `cardano-base` (and/or `cardano-ledger`), where `updateKES` is defined.

Reply:
Even if we don't create the new type, please rename the function.